### PR TITLE
Use temp binlog index replace binlog in querynode

### PR DESCRIPTION
**Pull request by cqy123456**
issue: https://github.com/milvus-io/milvus/issues/27678
When querynode load vector binlog for a sealed segment,  binlog fields will be replaced by a temperate binlog index for better search performance. And this binlog index will be released after loading vector index.
However, binlog needs to meet the following conditions:
1.  tempIndex = true in milvus.yaml;
2. user call create_index api, and build segment index task is not finish in indexnode or indexpool;
3.  index type != flat;

### CODE REVIEW

File: configs/milvus.yaml [hunk 1]
```diff
@@ -261,10 +261,10 @@ queryNode:
     # This parameter is only useful when enable-disk = true.
     # And this value should be a number greater than 1 and less than 32.
     chunkRows: 1024 # The number of vectors in a chunk.
-    growing: # growing a vector index for growing segment to accelerate search
+    tempIndex: # build a vector temperate index for growing segment or binlog to accelerate search
```
Comment (jiaoew1991) [inline configs/milvus.yaml:264]: I do not think `temp` is a good name 🤔

File: internal/querynodev2/segments/load_field_data_info.go [hunk 2]
```diff
@@ -64,6 +64,24 @@ func (ld *LoadFieldDataInfo) appendLoadFieldDataPath(fieldID int64, binlog *data
...
+func (ld *LoadFieldDataInfo) appendIndexParams(fieldID int64, indexParams map[string]string) error {
+	cFieldID := C.int64_t(fieldID)
+	for indexKey, indexValue := range indexParams {
+		cIndexKey := C.CString(indexKey)
+		defer C.free(unsafe.Pointer(cIndexKey))
+
+		cIndexValue := C.CString(indexValue)
+		defer C.free(unsafe.Pointer(cIndexValue))
+
+		status := C.AppendLoadFieldDataIndexParam(ld.cLoadFieldDataInfo, cFieldID, cIndexKey, cIndexValue)
```
Comment (xige-16) [inline internal/querynodev2/segments/load_field_data_info.go:76]: Is it possible to append indexparams only once when creating C.Collection?

File: configs/milvus.yaml (hunk 1 above)
Comment (cqy123456) [reply configs/milvus.yaml]: update “interimIndex”

**Comment by cqy123456** [comment]
rerun ut

**Comment by cqy123456** [comment]
rerun ut

**Pull request merged**